diff --git a/fs/proc/task_mmu.c b/fs/proc/task_mmu.c
index 0000000..1111111 100644
--- a/fs/proc/task_mmu.c
+++ b/fs/proc/task_mmu.c
@@ -19,6 +19,10 @@
 #include <linux/mm.h>
 #include <linux/mm_inline.h>
 #include <linux/ctype.h>
 #include <linux/sched/signal.h>
+#if defined(CONFIG_KSU_SUSFS_SUS_KSTAT) || defined(CONFIG_KSU_SUSFS_SUS_MAP)
+#include <linux/susfs_def.h>
+#endif

 #include <asm/elf.h>
 #include <asm/tlb.h>
diff --git a/kernel/sys.c b/kernel/sys.c
index 0000000..1111111 100644
--- a/kernel/sys.c
+++ b/kernel/sys.c
@@ -1237,6 +1237,10 @@ static int override_release(char __user *release, size_t len)
 	return ret;
 }

+#ifdef CONFIG_KSU_SUSFS_SPOOF_UNAME
+extern void susfs_spoof_uname(struct new_utsname* tmp);
+#endif
 SYSCALL_DEFINE1(newuname, struct new_utsname __user *, name)
 {
 	struct new_utsname tmp;
@@ -1246,6 +1250,9 @@ SYSCALL_DEFINE1(newuname, struct new_utsname __user *, name)
 	down_read(&uts_sem);
 	memcpy(&tmp, utsname(), sizeof(tmp));
+#ifdef CONFIG_KSU_SUSFS_SPOOF_UNAME
+	susfs_spoof_uname(&tmp);
+#endif
 	if (!strncmp(current->comm, "bpfloader", 9) ||
 	    !strncmp(current->comm, "netbpfload", 10) ||
 	    !strncmp(current->comm, "netd", 4) ||
